pkg/providers/aws/provider_redis.go

```diff
@@ -308,6 +308,20 @@ func (p *RedisProvider) createElasticacheCluster(ctx context.Context, r *v1alpha
 			errMsg := fmt.Sprintf("failed to add tags to elasticache: %s", msg)
 			return nil, types.StatusMessage(errMsg), errorUtil.Wrap(err, errMsg)
 		}
+		cacheClusterOutput, err := cacheSvc.DescribeCacheClusters(&elasticache.DescribeCacheClustersInput{
```

Comment:
Actually, just looking at this, we already Describe Cache Cluster on line 266. Probably worth moving, setting the `r.Status.Version` up there, using that output.

Reply:
yea seen that was worried that 
```golang
		cacheClustersOutput, err := cacheSvc.DescribeCacheClusters(&elasticache.DescribeCacheClustersInput{})
```
would not return the same as 
```golang
		cacheClusterOutput, err := cacheSvc.DescribeCacheClusters(&elasticache.DescribeCacheClustersInput{
					CacheClusterId: cache.CacheClusterId,
		})
		
```
will debug and find out

Reply:
In the loop 
```
for _, checkedCluster := range cacheClustersOutput.CacheClusters {
		if *checkedCluster.ReplicationGroupId == *foundCache.ReplicationGroupId {
			replicationGroupClusters = append(replicationGroupClusters, *checkedCluster)
		}
	}
```
The checkedCluster in the if statement, should be the same CacheCluster you are returning with 
```
cacheClusterOutput, err := cacheSvc.DescribeCacheClusters(&elasticache.DescribeCacheClustersInput{
					CacheClusterId: cache.CacheClusterId,
		})
```

Something like : 
```
	for _, checkedCluster := range cacheClustersOutput.CacheClusters {
		if checkedCluster.ReplicationGroupId != nil && foundCache.ReplicationGroupId != nil {
			if *checkedCluster.ReplicationGroupId == *foundCache.ReplicationGroupId {
				replicationGroupClusters = append(replicationGroupClusters, *checkedCluster)
				
				if checkedCluster.EngineVersion != nil && r.Status.Version != *checkedCluster.EngineVersion {
					r.Status.Version = *checkedCluster.EngineVersion
				}
			}
		}
	}
```